io.catenax.id_based_comment/1.0.0/IdBasedComment.ttl

```diff
@@ -0,0 +1,156 @@
+#######################################################################
+# Copyright (c) 2023 BASF SE
+# Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+# Copyright (c) 2023 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V. (represented by Fraunhofer IML)
+# Copyright (c) 2023 Henkel AG & Co. KGaA
+# Copyright (c) 2023 Mercedes Benz AG
+# Copyright (c) 2023 SAP SE
+# Copyright (c) 2023 SupplyOn AG
+# Copyright (c) 2023 Volkswagen AG
+# Copyright (c) 2023 ZF Friedrichshafen AG
+# Copyright (c) 2023 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This work is made available under the terms of the
+# Creative Commons Attribution 4.0 International (CC-BY-4.0) license,
+# which is available at
+# https://creativecommons.org/licenses/by/4.0/legalcode.
+#
+# SPDX-License-Identifier: CC-BY-4.0
+#######################################################################
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.0.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.0.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.0.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.0.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.id_based_comment:1.0.0#> .
+
+:IdBasedComment a samm:Aspect ;
+   samm:preferredName "Id Based Comment"@en ;
+   samm:description "Aspect model for an exchange of comment belongig to a entity."@en ;
+   samm:properties ( :commentId :objectId [ samm:property :author; samm:optional true ] [ samm:property :postedAt; samm:optional true ] [ samm:property :changedAt; samm:optional true ] [ samm:property :commentText; samm:optional true ] [ samm:property :commentType; samm:optional true ] [ samm:property :requestDelete; samm:optional true ] [ samm:property :listOfReferenceDates; samm:optional true ] :objectType ) ;
```

Comment:
This model is lacking `customer` and `supplier`. However, they are crucial for this model to be sound. Let me elaborate:

A customer-supplier relationship is a directed one: One party is customer, one is supplier. Comments, as well as the Material Demands and Capacity Groups they refer to, have IDs. These IDs are, in their description, explicitly defined to only be unique "within the business relationship between a customer and its supplier". Hence, to uniquely resolve a commentId, as well as an (objectId, objectType)-Tuple, you need to also know the supplier and the customer.

It is not possible to infer this information from the message header, as the message header only contains information about the sender and the receiver of the message. However, as both customer and supplier can send (create/update/delete) comments, it is not possible to know if the sender of the message is the customer or the supplier. Hence, it is not possible to derive the correct customer-supplier relationship. Hence, we can't uniquely address the comment, material demand or capacity group.

Example: As per our standard, it is valid that two companies A and B are both supplier and customer to each other -- so there are two business partner relationships (customer, supplier): (A, B) and (B, A). In both relationships, a material demand can be created with the same UUID. B creates a Material Demand for A with ID `123`, and A creates a Material Demand for B with ID `123`. The standard, today, forces us to not throw an error but treat those as separate Material Demands (because the ID must be treated as unique only _within_ the relationship, not globally). Now, A posts a comment on `objectId` `123`. Which Material Demand should receive the comment? A surely is the sender and B the receiver of the message, but from that you can't derive whether the business partner relationship (A, B) or (B, A) is targeted, because A could have commented on the Material Demand where it is supplier, or on the one where it is customer.

I know this is an edge case, but we allow (and must allow) sender-controlled IDs, and the other standards define them as not globally unique (allowing us to fail when we receive a second material demand (or comment/capacity group) with the same UUID but a different business partner relationship), but uniquely valid only within the business partner relationship (forcing us to accept them). Thus, the standards must also deal with this edge case.

Reply:
properties were added as discussed in the architecture meeting today